src/Constraint/Type/IntegerNumberValidator.php

```diff
@@ -22,6 +22,11 @@ public function validate($value, Constraint $constraint): void
             return;
         }
 
+        // if value is float, force to string. This will cast 5.0 => '5', and 5.5 => '5.5'
+        if (is_float($value)) {
```

Comment:
Instead of only casting to string when the value is of type float, you can always do that and check on ```((string)(int)$value !== (string)$value)``` on line 35/40 and add the ```is_float($value)``` (with ```&& is_float($value) === false```) to line 26/31. ;)